### ISSUE TITLE
[cmake] libminiz.so not part of the install target

### ISSUE THREAD

**Issue by ThomasDevoogdt** [fluent/fluent-bit#6711]
I'm working to upstream fluent-bit as a [buildroot](https://buildroot.org/) package: https://patchwork.ozlabs.org/project/buildroot/patch/20230120071043.1169953-1-thomas.devoogdt@barco.com/

It seems that `libminiz*.so*` is not installed on the target using the default buildroot `cmake-package` system. (Basically, `cmake . && make && make install`)

Used Config:
```
FLUENT_BIT_CONF_OPTS += \
	-DFLB_DEBUG=No \
	-DFLB_RELEASE=Yes \
	-DFLB_EXAMPLES=No \
	-DFLB_CHUNK_TRACE=No \
	-DFLB_BACKTRACE=No \
	-DFLB_LUAJIT=No
```

So can this be added, so that I can drop the custom target hook:? 
```
cp -dpf $(@D)/lib/libminiz*.so* $(TARGET_DIR)/usr/lib/
```

While at it, is it possible to provide a custom target path, so that configs end up in `/etc/` rather than `/usr/etc/`? (This question aside)

**Comment by patrick-stephens** [comment]
Any chance you can submit a PR for this? We'll need to verify it's ok for all supported targets but that'll be a lot quicker than waiting for someone else to pick it up.

**Comment by ThomasDevoogdt** [comment]
I can try to create a patch. The cmake files looked quite complicated at first inside. Otherwise I would already have opened a pull request. If you can provide a quick an dirty patch, then I'm willing to apply and debug it.

**Comment by ThomasDevoogdt** [comment]
Update (related to the question aside), the `/usr/etc/` path can be replaced by setting `CMAKE_INSTALL_SYSCONFDIR`, added this:
```
# Move the config files from /usr/etc/ to /etc/.
FLUENT_BIT_CONF_OPTS += \
	-DCMAKE_INSTALL_SYSCONFDIR="/etc/"
```
Fixing the libminiz is more difficult. But I guess that it is related to buildroot which sets BUILD_SHARED_LIBS globally, so Miniz also takes that config option, while it shouldn't.

**Comment by patrick-stephens** [comment]
Yeah we set the variables during the package builds, e.g. https://github.com/fluent/fluent-bit/blob/16eae107868bf03da2fce6ca4acd3f6ab2d3c4ff/packaging/distros/ubuntu/Dockerfile#L139

**Comment by ThomasDevoogdt** [comment]
Fluent-bit is in the meantime added to buildroot: https://github.com/buildroot/buildroot/commit/6a0f7c39bcb48fc13aa2ce3fc4996baf1be66483.
(Original repo: https://git.busybox.net/buildroot/commit/?id=6a0f7c39bcb48fc13aa2ce3fc4996baf1be66483)

The `libminiz*.so*` bundling problem is fixed by forcing all compiled files to be static:
```
FLUENT_BIT_CONF_OPTS += \
	-DBUILD_SHARED_LIBS=OFF
```

It is still a valid problem, so I keep the issue open. But I believe we found a clean way to add it in Buildroot.

**Comment by ThomasDevoogdt** [comment]
This has been fixed by bumping miniz in commit 47fd29ae7b9c393357e318831e0bf616c7856cde, and more specific by commit https://github.com/richgel999/miniz/commit/0ce345c44ad1852f29db12c1eb65a429038b2809.
